test/integration/chronokvs/chronokvs_integration_test.cpp

```diff
@@ -0,0 +1,334 @@
+        std::cout << "  Purpose: Verify ChronoKVS API functionality for GitHub Actions" << std::endl;
+        std::cout << "  Tests:   Put Operations, Data Propagation, Get History, Get Operations" << std::endl;
+        std::cout << std::string(80, '=') << std::endl;
+
+        bool test1_result = false;
+        bool test2_result = false;
+        bool test3_result = false;
+
+        // Test 1: Put
+        test1_result = test1_put();
+
+        // Wait for data propagation (only if Test 1 passed)
+        if(test1_result)
+        {
+            waitForDataPropagation();
+        }
+        else
+        {
+            printHeader("DATA PROPAGATION WAIT");
+            std::cout << "  ⚠️  Skipping data propagation wait due to Test 1 failure" << std::endl;
+            std::cout << "  Subsequent tests may not have data to work with" << std::endl;
+        }
+
+        // Test 2: Get History
+        test2_result = test2_get_history();
+
+        // Test 3: Get (only if we have data from previous tests)
+        if(test1_result && timestamps.size() > 0)
+        {
+            test3_result = test3_get();
+        }
+        else
+        {
+            printHeader("TEST 3: GET OPERATIONS");
+            std::cout << "  ⚠️  Skipping Test 3 (Get) - no data available from Test 1" << std::endl;
+            std::cout << "  This test requires successful put operations to proceed" << std::endl;
+            test3_result = false;
+        }
+
+        // Final summary
+        printHeader("FINAL TEST RESULTS");
+        std::cout << "  Test 1 (Put):        " << (test1_result ? "✅ PASSED" : "❌ FAILED") << std::endl;
+        std::cout << "  Test 2 (Get History): " << (test2_result ? "✅ PASSED" : "❌ FAILED") << std::endl;
+        std::cout << "  Test 3 (Get):        " << (test3_result ? "✅ PASSED" : "❌ FAILED") << std::endl;
+        printSeparator();
+
+        int passed_tests = (test1_result ? 1 : 0) + (test2_result ? 1 : 0) + (test3_result ? 1 : 0);
+        std::cout << "  Overall: " << passed_tests << "/3 tests passed" << std::endl;
+
+        if(passed_tests == 3)
+        {
+            std::cout << "\n  🎉 ALL CHRONOKVS API TESTS PASSED! 🎉" << std::endl;
+            std::cout << "  The ChronoKVS integration is working correctly and ready for GitHub Actions." << std::endl;
+        }
+        else if(passed_tests > 0)
+        {
+            std::cout << "\n  ⚠️  PARTIAL SUCCESS: " << passed_tests << "/3 tests passed" << std::endl;
+            std::cout << "  Some ChronoKVS functionality is working, but there may be issues to investigate."
+                      << std::endl;
+        }
+        else
+        {
+            std::cout << "\n  ❌ ALL TESTS FAILED" << std::endl;
+            std::cout << "  ChronoKVS integration has critical issues that need to be resolved." << std::endl;
+        }
+
+        std::cout << std::string(80, '=') << std::endl;
+
+        return passed_tests == 3;
+    }
+};
+
+int main()
```

Comment:
I think we need to allow passing client conf file to provide contact point of Visor. Otherwise, it is too limited to test or run anything in the distributed deployment.